esrally/paths.py

```diff
@@ -41,6 +41,10 @@ def install_root(cfg=None):
     return os.path.join(races_root(cfg), install_id)
 
 
+def libs():
+    return os.path.join(rally_confdir(), "lib")
```

Comment:
nit: this should probably be called "libs" or "dependencies" not just "lib".

The issue in putting this in ~/.rally/lib is that it will affect all tracks and will cause havoc when multiple tracks start using this feature. The other options I see are the track directory (but we need to tell git to ignore the libs) and the race directory.

Reply:
The code fully wipes and re-creates the directory before `sys.path.insert(0)` to avoid collisions with other dependencies

Reply:
Oh, I see, not sure how I missed it. Can we also document this?